### PR TITLE
Move event hooks to markdig extension setup method

### PR DESCRIPTION
**Pull request by yufeih**
This allows extensions to be applied to `MarkdownPipelineBuilder` without calling `UseXXX`.

### CODE REVIEW

File: src/Microsoft.DocAsCode.MarkdigEngine.Extensions/LineNumber/LineNumberExtension.cs [hunk 1]
```diff
@@ -3,23 +3,34 @@
...
             {
-                AddSourceInfoInDataEntry(document, filePath);
+                AddSourceInfoInDataEntry(document, _context.GetFilePath(_context.File));
             };
         }
 
+        public void Setup(MarkdownPipeline pipeline, IMarkdownRenderer renderer)
+        {
+
+        }
```
Comment (OsmondJiang) [inline src/Microsoft.DocAsCode.MarkdigEngine.Extensions/LineNumber/LineNumberExtension.cs]: what's that for?

File: src/Microsoft.DocAsCode.MarkdigEngine.Extensions/TabGroup/TabGroupExtension.cs [hunk 2]
```diff
@@ -10,6 +10,21 @@ public class TabGroupExtension : IMarkdownExtension
...
         public void Setup(MarkdownPipelineBuilder pipeline)
         {
+            var tabGroupAggregator = new TabGroupAggregator();
+            var aggregateVisitor = new MarkdownDocumentAggregatorVisitor(tabGroupAggregator);
+
+            var tagGroupIdRewriter = new TabGroupIdRewriter();
+            var tagGroupIdVisitor = new MarkdownDocumentVisitor(tagGroupIdRewriter);
+
+            var activeAndVisibleRewriter = new ActiveAndVisibleRewriter();
+            var activeAndVisibleVisitor = new MarkdownDocumentVisitor(activeAndVisibleRewriter);
```
Comment (OsmondJiang) [inline src/Microsoft.DocAsCode.MarkdigEngine.Extensions/TabGroup/TabGroupExtension.cs]: >var activeAndVisibleVisitor = new MarkdownDocumentVisitor(activeAndVisibleRewriter); [](start = 12, length = 84)

var activeAndVisibleVisitor = new MarkdownDocumentVisitor(new ActiveAndVisibleRewriter());

Comment (yufeih) [reply src/Microsoft.DocAsCode.MarkdigEngine.Extensions/TabGroup/TabGroupExtension.cs]: This is a copy of old code. I do not intend to change existing styles.

File: src/Microsoft.DocAsCode.MarkdigEngine.Extensions/LineNumber/LineNumberExtension.cs (hunk 1 above)
Comment (yufeih) [reply src/Microsoft.DocAsCode.MarkdigEngine.Extensions/LineNumber/LineNumberExtension.cs]: in v2, filePath is a `RelativePath`, `RelativePath.ToString()` starts with '~/' if the path is from working directory. The expected source info should not contain '~/'

**Pull request merged**